fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/fluid/DefaultFluidRenderer.java

```diff
@@ -0,0 +1,41 @@
+package net.fabricmc.fabric.impl.client.rendering.fluid;
+
+import net.fabricmc.fabric.api.client.render.fluid.v1.CustomFluidRenderer;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.render.VertexConsumer;
+import net.minecraft.client.render.block.FluidRenderer;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockRenderView;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+public class DefaultFluidRenderer implements CustomFluidRenderer {
+	public static final DefaultFluidRenderer INSTANCE = new DefaultFluidRenderer();
+
+	// Lazily loaded because FluidRenderer might not yet exist at instantiation time
+	private FluidRenderer defaultRenderer;
```

Comment:
You need this to be `volatile` for double-checked locking to be safe.

Reply:
This entire class is removed now